Dev 1555: Add Open Banking developer docs to API Reference (#587)
Dwolla/dev-portal: pages/docs/balance/api-reference/open-banking/index.mdx

---
layout: guides
product: balance
title: "Overview"
group:
  title: Open Banking
  weight: 12
guide:
  step: 0
meta:
  title: "Dwolla API Reference | Open Banking: Overview"
  description: "Dwolla Exchange Sessions API connects your app, open banking providers and user banks. Streamline onboarding with instant account verification and leverage Dwolla's pre-integrated solutions."
---

# Open Banking

This section dives into the process of leveraging Dwolla's Exchange Sessions API for Instant Account Verification (IAV) within your application. Dwolla's Exchange Sessions API enables you to leverage open banking providers to perform bank account verification in real-time, streamlining your onboarding process and improving user experience. An Exchange Session represents a generated IAV session that is established between Dwolla and integrated open banking service providers for the purposes of user bank account verification.

### Instant Account Verification Flow

Here's a breakdown of the steps:

> 1. Initiate Exchange Session

Reviewer: ```suggestion
**Step 1: Initiate Exchange Session**
```

- Use the [create an exchange session](/docs/balance/api-reference/exchanges/create-exchange-for-a-customer) endpoint of the Dwolla API to initiate an Exchange Session for a Customer.
- Specify the desired open banking provider (e.g., MX) within the request body.

> 2. Retrieve Exchange Session and Complete IAV Flow

Reviewer: ```suggestion
**Step 2: Retrieve Exchange Session and Complete IAV Flow**
```

> - Make an API call to [retrieve an exchange session](/docs/balance/api-reference/open-banking/retrieve-an-exchange-session). Upon success, Dwolla will return a `external-provider-session` URL where your application will redirect the user to a secure login.

Reviewer: ```suggestion
- Make an API call to [retrieve an exchange session](/docs/balance/api-reference/open-banking/retrieve-an-exchange-session). Upon success, Dwolla will return an `external-provider-session` URL where your application will redirect the user to a secure login.
```

- The user logs in to their bank account and grants your application permission to access specific financial data for account verification.
- Implement proper error handling mechanisms to address potential issues during the verification process.

> 3. Data Retrieval and Exchange Creation

Reviewer: ```suggestion
**Step 3: Data Retrieval and Exchange Creation**
```

> - Upon successful authorization, you'll obtain specific user and selected account details from the open banking provider which will be used to [create an exchange](/docs/balance/api-reference/exchanges/create-exchange-for-an-external-party).

Reviewer: ```suggestion
- Upon successful authorization, you'll obtain specific user and selected account details from the open banking provider which will be used to [create an exchange](/docs/balance/api-reference/exchanges/create-exchange-for-a-customer).
```

- Dwolla retrieves relevant account data from the user's bank through the secure open banking provider API connection established during the Exchange Session.

> 4. Funding Source Creation and Account Verification

Reviewer: ```suggestion
**Step 4: Funding Source Creation and Account Verification**
```

- Using the [create a funding source endpoint](/docs/balance/api-reference/funding-sources/create-funding-source-for-customer), you'll create a funding source by passing in the exchange resource from the previous step.

> - Dwolla returns a 201 response and triggers `customer_funding_source_added` and `customer_funding_source_verified` webhooks indicating the funding source and account verification is complete.

Reviewer: ```suggestion
- Dwolla returns a 201 response and triggers `customer_funding_source_added` and `customer_funding_source_verified` webhooks indicating the funding source creation and verification is complete.
```

### Exchange Sessions resource

| Parameter | Description                                                                                       |
| --------- | ------------------------------------------------------------------------------------------------- |

> | \_links   | A unique string value which can be used to identify a third-party “partner” via the Exchange API. |

Reviewer: We should probably modify the description for `_links` since it includes additional relevant links to the `exchange-partner`, `external-provider-session`. 

We should probably include an `id` key:value in the API as well.

Author: Agreed! I think that makes sense. Add the id to be consistent across all other resources
